### ISSUE TITLE
[Bug] keychron/q1v1/ansi keyboard bricks the keyboard

### ISSUE THREAD

**Issue by LyraelRayne** [Keychron/qmk_firmware#212]
### Describe the Bug

Keyboard doesn't work if you flash q1v1 firmware from master.

"Doesn't work" means:
- initially no keys work and rgb is off
- after some time rgb slowly pulses red but not smoothly (like low-fps graphics)
- Key inputs eventually work but with very low polling rate

For those coming from the internet, the 9 month old `keychron-q1v1-q1v2` branch does work as expected (use keyboard keychron/q1)

Commit 38f89dfc2f8535ca8721a9ab91db850e78273e2f is the first broken commit.

### Keyboard Used

keychron/q1v1/ansi

### Operating System

Windows

### qmk doctor Output

```text
Ψ QMK Doctor is checking your environment.
Ψ CLI version: 1.1.2
Ψ QMK home: C:/qmk/qmk_firmware
Ψ Detected Windows 10 (10.0.22631).
Ψ QMK MSYS version: 1.8.0
Ψ Userspace enabled: False
Ψ Git branch: kc_master
Ψ Repo version: 0.14.29
⚠ Git has unstashed/uncommitted changes.
Ψ - Latest kc_master: 2023-12-18 13:53:02 +1100 (9539f135d8) -- Remove obvious user keymaps, `keyboards/[0-9]*` edition. (#22691)
Ψ - Latest upstream/master: 2024-01-06 02:06:14 +0100 (a474cac2c3) -- Update keymap.md (#22834)
Ψ - Latest upstream/develop: None
Ψ - Common ancestor with upstream/master: 2023-12-18 13:53:02 +1100 (9539f135d8) -- Remove obvious user keymaps, `keyboards/[0-9]*` edition. (#22691)
Ψ - Common ancestor with upstream/develop: None
Ψ All dependencies are installed.
Ψ Found arm-none-eabi-gcc version 12.2.0
Ψ Found avr-gcc version 12.2.0
⚠ We do not recommend avr-gcc newer than 8. Downgrading to 8.x is recommended.
Ψ Found avrdude version 7.0
Ψ Found dfu-programmer version 1.1.0
Ψ Found dfu-util version 0.11
Ψ Submodules are up to date.
Ψ Submodule status:
Ψ - lib/chibios: 2023-04-15 13:48:04 +0000 --  (11edb1610)
Ψ - lib/chibios-contrib: 2023-07-17 11:39:05 +0200 --  (da78eb37)
Ψ - lib/googletest: 2021-06-11 06:37:43 -0700 --  (e2239ee6)
Ψ - lib/lufa: 2022-08-26 12:09:55 +1000 --  (549b97320)
Ψ - lib/vusb: 2022-06-13 09:18:17 +1000 --  (819dbc1)
Ψ - lib/printf: 2022-06-29 23:59:58 +0300 --  (c2e3b4e)
Ψ - lib/pico-sdk: 2023-02-12 20:19:37 +0100 --  (a3398d8)
Ψ - lib/lvgl: 2022-04-11 04:44:53 -0600 --  (e19410f8)
Ψ QMK is ready to go, but minor problems were found
```

### Is AutoHotKey / Karabiner installed

- [ ] AutoHotKey (Windows)
- [ ] Karabiner (macOS)

**Comment by LyraelRayne** [comment]
Compiling with rgb disabled causes everything but RGB to work so it's clearly related to the RGB code

**Comment by LyraelRayne** [comment]
I have tried compiling using older QMK_MSYS (with v8 of avr-gcc) but it still bricks

**Comment by LyraelRayne** [comment]
Fixed upstream in qmk/qmk_firmware@0b0c31665e7b4e5a9cd9ac3ac94b0fba10514aa3
